File: 프로그래머스/3/43238. 입국심사/입국심사.py
def solution(n, times):
    start, end = 0, times[-1] * n
    while start < end:
        mid = (start + end) // 2
        
        cnt = 0
        for t in times:
            cnt += mid // t
        
        if cnt < n:
            start = mid + 1
        else:
            end = mid
    return end

File: 프로그래머스/3/43238. 입국심사/입국심사.java
class Solution {
    public long solution(int n, int[] times) {
        long start = 0;
        long end = (long) times[times.length - 1] * n;
        
        while (start < end) {
            long mid = (start + end) / 2;
            
            long cnt = 0;
            for (int t : times) {
                cnt += mid / t;
            }
            
            if (cnt < n) {
                start = mid + 1;
            } else {
                end = mid;
            }
        }
        return end;
    }
}